### PR TITLE
[v4.4.1-crio] Bump c/common to v0.52.0

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
As the title says.  This replaces
https://github.com/containers/podman/pull/22355.

I'm not sure how I managed, but I accidentally pushed the original PR, #22355, into the v4.4.1-crio branch as a merge commit.

I've reverted that, and redid the vendor for c/common 0.52.0

This dragged in c/image v5.25.0, and I also needed to bump image builder as the version of Docker that also got dragged in dropped the LCOWSupported functionality and c/storage by that time had taken it over.

So long story short, a lot more changes than I'm entirely comfortable with.

@kwilczynski PTAL.  Sorry about the very slow response on this, we had to get some CVE fixes first, and in so doing, discovered the CI in this branch was pretty badly broken and had to fix that .

Addresses: https://issues.redhat.com/browse/OCPBUGS-32144

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by kwilczynski** [comment]
@TomSweeneyRedHat, thank you for tending to this. Appreciated!

It seems that some of the checks are failing in the post-steps. I am not sure why, though. Some older script versions, perhaps?

**Comment by mheon** [comment]
Build failures are legitimate. Completions errors?

**Comment by TomSweeneyRedHat** [comment]
Repushed, with my learning for the day.  @edsantiago and his eagle eye spotted the issue in the voluminous log.  Given the change in the cobra that was vendored in, I needed to  `make complete`, add the changes, rebase, and repush.  Fingers crossed!  Thanks again for the assist Ed, much appreciated.

**Comment by TomSweeneyRedHat** [comment]
@mheon I'm not surprised, this is bloated, do you want to slap a tag on this?

**Comment by TomSweeneyRedHat** [comment]
@cevich, what are your thoughts on this one? I had to turn off some lint checks and, more importantly, move to a new capabilities check. I'm not sure if that last change or one of the many new files is asking for a Golang bump.

Do you have any thoughts on the next steps here?

**Comment by cevich** [comment]
> I had to turn off some lint checks and, more importantly, move to a new capabilities check. I'm not sure if that last change or one of the many new files is asking for a Golang bump.

Bumping golang on an old release branch is always a big risk for overall CI-health as you and I have been through countless times at this point.  I'm assuming you tried the suggested vendoring fix.  Is that what is suggesting the golang bump?

**Comment by kwilczynski** [comment]
I think, we are reaching a point where we need to decide whether we need this branch and the bump. This has become a bit of a headache to maintain.

What do you think, @haircommander?

**Comment by haircommander** [comment]
I don't necessarily think we need fedora rpm builds for these branches. If we continue to package this branch in openshift we can use that build process to catch issues, as these builds will never make it into fedora (or any real version of rhel either). what do folks think?

**Comment by TomSweeneyRedHat** [comment]
As this has been idle for a while, and some question over need, closing.
